### PR TITLE
Secret management API- followup 

### PR DESCRIPTION
**Pull request by SiniWilson**
### Requirements

* Minor bug fixes
* Adding vault configuration

### Description of the Change

- Introduced a new key in swagger as part of the role details.

### CODE REVIEW

File: core/jazz_t-vault/swagger/swagger.json [hunk 1]
```diff
@@ -3394,6 +3394,13 @@
                     "type": "string",
                     "description": "Auth type of the role"
                 },
+                "iamPrincipleArn": {
```
[the remark this answers is not in the document]
Comment (devsatishm) [reply core/jazz_t-vault/swagger/swagger.json]: Role should have one ARN. Why is it an array in the first place?

[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_t-vault/swagger/swagger.json]: It came as an array in the tvault response. So I used the same here. Please let me know, need to change that

[the remark this answers is not in the document]
Comment (raghits) [reply core/jazz_t-vault/swagger/swagger.json]: @SiniWilson Does the t-vault API return an array of ARN of the principals (Users)?

[the remark this answers is not in the document]
Comment (raghits) [reply core/jazz_t-vault/swagger/swagger.json]: Let us just keep it as an array and call it `arns`. Also, make sure in the code the variables are named accordingly i.e. roleDetail.arns etc.

[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_t-vault/swagger/swagger.json]: Done

**Pull request merged**